samplers.qmd

```diff
@@ -121,7 +121,216 @@ We can see from the autocorrelation function that we have to wait almost 100 ste
 
 ### Adaptive Metropolis-Hastings Sampler
 
-TODO: Content about the adaptive Metropolis-Hastings sampler to be placed here.
+**Overview**  
+This sampler extends the basic (non-adaptive) random-walk Metropolis-Hastings algorithm by dynamically updating its proposal distribution as the chain progresses. In a standard random-walk MCMC, one must guess a single variance-covariance matrix (VCV) up front. If the target distribution is high-dimensional or has strong correlations, it can be challenging to find a VCV that allows the chain to explore efficiently.
+
+With this **adaptive** approach, the VCV and the overall scaling factor evolve according to the observed samples. The sampler “learns” the covariance structure of the target distribution on-the-fly, leading to (potentially) much lower autocorrelation, better mixing, and more efficient exploration. Formally, this adaptivity modifies the Markov property, so care must be taken to ensure the adaptation “diminishes” over time.  
+
+**Main ideas**
+
+1. **Adaptive Shaping**  
+   - After each iteration, the algorithm updates an **empirical VCV** using the newly accepted parameter values.  
+   - Early samples are gradually *forgotten* (depending on `forget_rate` and `forget_end`) so that the empirical VCV primarily reflects the more recent part of the chain. This is helpful if the chain starts far from the high posterior density region - early samples will be informative about the shape of movement towards this region but once there will not be informative about the shape of the region itself.
+   - In the proposal kernel, the empirical VCV is weighted according to its sample size against an initial estimate of the VCV, `initial_vcv`, with weight `initial_vcv_weight`.
+
+2. **Adaptive Scaling**  
+   - The sampler also updates a **global scaling factor** (which is squared and multiplied by $2.38^2 / d$ in the proposal, where $d$ is the number of parameters) to target a desired acceptance rate (`acceptance_target`). The optimal scaling factor should be 1 in a Gaussian target space when using the true underlying VCV.
+   - If acceptance is too high, the sampler increases the scaling factor. If acceptance is too low, it decreases it. Doing this in log-space is often more stable (`log_scaling_update = TRUE`).
+
+3. **Adaptation Stopping**  
+   - Eventually, the sampler stops adapting (`adapt_end`), freezing the proposal distribution for the remainder of the run. This helps restore formal MCMC convergence properties, as purely adaptive samplers can break standard theory if they keep adapting indefinitely.
+
+**Citation**  
+This implementation is inspired by an “accelerated shaping” adaptive algorithm from Spencer (2021):
+
+> Spencer SEF (2021) Accelerating adaptation in the adaptive Metropolis–Hastings random walk algorithm. *Australian & New Zealand Journal of Statistics*, 63:468–484.
+
+---
+
+::: {.callout-note}
```

Comment:
This is very long and not all readers will be interested in the details - this should be made collapsible I think.

Reply:
I thought so too, done!